onnxruntime/core/optimizer/common_subexpression_elimination.cc

```diff
@@ -260,20 +269,52 @@ bool EquivalenceClass::operator==(const EquivalenceClass& other) const {
     return true;
   }
 
+  bool are_input_same = inputs_.size() == other.inputs_.size();
+  
+  if(are_input_same) {
+    for(unsigned long i=0; i<inputs_.size(); i++) {
```

Comment:
Note that currently an invariant is maintained that we never create two distinct but equal EquivalenceClasses. For operations, this is done by looking up a potentially new EquivalenceClass in `value_to_representative` map. You should do the same for initializers. Then this loop is not required.
Commutative operations can be handled by sorting the arguments in EquivalenceClass constructor (`Normalize` function should be a good place for that). Sorting by address is sufficient due to the above invariant.

Reply:
@MaximKalininMS, regarding the sorting of inputs - will it work for operations like `Div` if we always sort the inputs?

Reply:
Inputs should be sorted only for commutative operations (for which you want to merge `op(x, y)` and `op(y, x)`).